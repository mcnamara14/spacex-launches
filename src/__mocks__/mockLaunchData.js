export const mockLaunchData = [{
    "flight_number": 1,
    "mission_name": "FalconSat",
    "upcoming": false,
    "launch_year": "2006",
    "launch_date_unix": 1143239400,
    "launch_date_utc": "2006-03-24T22:30:00.000Z",
    "rocket": {
        "rocket_name": "Falcon 1",
        "first_stage": {
            "cores": [
                {
                    "reused": false,
                    "land_success": false,
                }
            ]
        },
    },
    "reuse": {
        "core": false,
        "side_core1": false,
        "side_core2": false,
        "fairings": false,
        "capsule": false
    },
    "links": {
        "mission_patch": "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
        "mission_patch_small": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
        "article_link": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
        "wikipedia": "https://en.wikipedia.org/wiki/DemoSat",
        "video_link": "https://www.youtube.com/watch?v=0a_00nJ_Y88"
    },
  },
  {
  "flight_number": 2,
  "mission_name": "FalconSun",
  "upcoming": false,
  "launch_year": "2008",
  "launch_date_unix": 1143239500,
  "launch_date_utc": "2008-03-24T22:30:00.000Z",
  "rocket": {
    "rocket_name": "Falcon 2",
    "first_stage": {
        "cores": [
            {
                "reused": true,
                "land_success": false,
            }
        ]
    },
  },
  "reuse": {
    "core": false,
    "side_core1": false,
    "side_core2": false,
    "fairings": false,
    "capsule": false
  },
  "links": {
    "mission_patch": "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
    "mission_patch_small": "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png",
    "article_link": "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html",
    "wikipedia": "https://en.wikipedia.org/wiki/DemoSat",
    "video_link": "https://www.youtube.com/watch?v=0a_00nJ_Y88"
  }
}];