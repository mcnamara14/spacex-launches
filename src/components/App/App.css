* {
  font-family: 'geomanistregular';
}

.App {
  width: 100vw;
  background-size: cover;
  background-attachment: fixed;
  padding-bottom: 30px;
}

main {
  width: 900px;
  min-height: 1000px;
  margin: 0 auto;
  text-align: center;
}

@font-face {
  font-family: 'geomanistregular';
  src: url('../../fonts/geomanist-regular-webfont.eot');
  src: url('../../fonts/geomanist-regular-webfont.eot?#iefix') format('embedded-opentype'),
       url('../../fonts/geomanist-regular-webfont.woff2') format('woff2'),
       url('../../fonts/geomanist-regular-webfont.woff') format('woff'),
       url('../../fonts/geomanist-regular-webfont.ttf') format('truetype'),
       url('../../fonts/geomanist-regular-webfont.svg#geomanistregular') format('svg');
  font-weight: normal;
  font-style: normal;
}