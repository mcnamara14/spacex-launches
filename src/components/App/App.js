import React, { Component } from 'react';
import './App.css';
import Header from '../Header/Header';
import LaunchCards from '../LaunchCards/LaunchCards';
import spaceBgImg from './images/background.jpg';

class App extends Component {
  render() {
    const bgImg = {backgroundImage: "url(" + spaceBgImg + ")"};

    return (
      <section className="App" style={bgImg}>
        <main>
          <Header />
          <LaunchCards />
        </main>
      </section>
    );
  }
}

export default App;
