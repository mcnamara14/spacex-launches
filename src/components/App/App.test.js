import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import { mockLaunchData, cleanMockLaunchData } from '../../__mocks__/mockLaunchData';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(mockLaunchData)
    }));
    
    wrapper = shallow(<App />, { disableLifecycleMethods: true });

    wrapper.instance().cleanLaunchData = jest.fn();
    wrapper.instance().storeLaunches = jest.fn();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('fetchLaunches', () => {
    it('should fetch with the correct argument', async () => {
      const expectedUrl = `https://api.spacexdata.com/v2/launches`
  
      await wrapper.instance().fetchLaunches();
  
      expect(window.fetch).toHaveBeenCalledWith(expectedUrl)
    });

    it('should call cleanLaunchData with correct data', async () => {
      const expected = mockLaunchData;

      await wrapper.instance().fetchLaunches();

      expect(wrapper.instance().cleanLaunchData).toHaveBeenCalledWith(expected)
    });
  })

  describe('cleanLaunchData', () => {
    it('should clean raw launch data', () => {
      wrapper = shallow(<App />, { disableLifecycleMethods: true });

      const result = wrapper.instance().cleanLaunchData(mockLaunchData);
  
      expect(result).toEqual(cleanMockLaunchData)
    })
  })

  describe('storeLaunches', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App clearLaunches={jest.fn()} storeLaunches={jest.fn()}/>, { disableLifecycleMethods: true });
      wrapper.instance().storeLaunches(cleanMockLaunchData);
    });

    it('should call clearLaunches in props', () => {
      expect(wrapper.instance().props.clearLaunches).toHaveBeenCalled()
    })

    it('should call storeLaunches with correct args', () => {
      const expected = (
        [
          "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", 
          "Falcon 2", 
          "Merlin B", 
          "03/24/2008", 
          "Engine failure at 33 seconds and loss of vehicle", 
          2, "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html", 
          false, 
          true, 
          false], 
          [
            "https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png", 
            "Falcon 1", 
            "Merlin A", 
            "03/24/2006", 
            "Engine failure at 33 seconds and loss of vehicle", 
            1, "https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html", 
            false, 
            false, 
            false]
      )
      expect(wrapper.instance().props.storeLaunches).toHaveBeenCalledWith(...expected)
    })
  })
});