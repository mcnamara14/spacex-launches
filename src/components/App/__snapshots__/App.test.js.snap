// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <img
                className="rocketBg"
                src="rocket.png"
        />
        <Header
                fetchLaunches={[Function]}
        />
        <Connect(LaunchCards) />
</main>,
      "className": "App",
      "style": Object {
        "backgroundImage": "url(background.jpg)",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="rocketBg"
            src="rocket.png"
/>,
          <Header
            fetchLaunches={[Function]}
/>,
          <Connect(LaunchCards) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rocketBg",
            "src": "rocket.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fetchLaunches": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <img
                    className="rocketBg"
                    src="rocket.png"
          />
          <Header
                    fetchLaunches={[Function]}
          />
          <Connect(LaunchCards) />
</main>,
        "className": "App",
        "style": Object {
          "backgroundImage": "url(background.jpg)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="rocketBg"
              src="rocket.png"
/>,
            <Header
              fetchLaunches={[Function]}
/>,
            <Connect(LaunchCards) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "rocketBg",
              "src": "rocket.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fetchLaunches": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
